File: flutter_app/lib/src/models/skill.dart
class Skill {
  final String id;
  final String name;
  final String description;
  final String category;
  final int level;
  
  Skill({
    required this.id,
    required this.name,
    required this.description,
    required this.category,
    required this.level,
  });
  
  factory Skill.fromJson(Map<String, dynamic> json) {
    return Skill(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      category: json['category'],
      level: json['level'],
    );
  }
  
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'category': category,
      'level': level,
    };
  }
}

class SkillCreate {
  final String name;
  final String description;
  final String category;
  final int? level;
  
  SkillCreate({
    required this.name,
    required this.description,
    required this.category,
    this.level,
  });
  
  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'description': description,
      'category': category,
      'level': level,
    };
  }
} 

File: api/src/models/skill.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;
use utoipa::ToSchema;

#[derive(Debug, Serialize, Deserialize, Clone, ToSchema)]
pub struct Skill {
    #[schema(example = "123e4567-e89b-12d3-a456-426614174000")]
    pub id: Uuid,
    #[schema(example = "Lockpicking")]
    pub name: String,
    #[schema(example = "Ability to open locks without a key")]
    pub description: Option<String>,
    #[schema(example = "STEALTH")]
    pub category: String,
    #[schema(example = 5)]
    pub level: i32,
}

#[derive(Debug, Deserialize, ToSchema)]
pub struct SkillCreate {
    #[schema(example = "Lockpicking")]
    pub name: String,
    #[schema(example = "Ability to open locks without a key")]
    pub description: Option<String>,
    #[schema(example = "STEALTH")]
    pub category: String,
    #[schema(example = 5)]
    pub level: Option<i32>,
}

#[derive(Debug, Serialize, ToSchema)]
pub struct SkillListResponse {
    pub skills: Vec<Skill>,
    pub total: i64,
    pub offset: i64,
    pub limit: i64,
} 